src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -142,6 +158,14 @@ private void SetEntry(CacheEntry entry)
                         // Adding the new entry will succeed only if no entry has been added since.
                         // This guarantees removing an old entry does not prevent adding a new entry.
                         entryAdded = _entries.TryAdd(entry.Key, entry);
+
+                        if (entryAdded)
+                        {
+                            // We have only added the difference between the new size and the old size.
+                            // Since the old one was removed, we need to add the old entry's size back
+                            // to ensure the total size of the new entry is accounted for.
+                            Interlocked.Add(ref _cacheSize, priorEntry.Size.Value);
```

Comment:
This could lead to a situation where the capacity is exceeded once SetEntry is complete. Will look for another way to accomplish this.